--- a/src/openalpr/support/filesystem.cpp
+++ b/src/openalpr/support/filesystem.cpp
@@ -149,6 +149,8 @@ namespace alpr
 
 #if defined(__APPLE__)
       double milliseconds = (stat_buf.st_ctimespec.tv_sec * 1000) +  (((double) stat_buf.st_ctimespec.tv_nsec) / 1000000.0);
+#elif defined(ANDROID)
+      double milliseconds = (stat_buf.st_ctime * 1000) +  (((double) stat_buf.st_ctime) / 1000000.0);
 #else
       double milliseconds = (stat_buf.st_ctim.tv_sec * 1000) +  (((double) stat_buf.st_ctim.tv_nsec) / 1000000.0);
 #endif
